### ISSUE TITLE
modifyObstructiveCode unintentionally broke code functionality. 

### ISSUE THREAD

**Issue by jennifer-shehane** [cypress-io/cypress#1454]
## How to reproduce:

Run the test below - the screen will stay on a loading screen instead of redirecting to the login page (as it does in Chrome). Every so often, a test will fail with a cross-origin script error claiming that it `Cannot read property 'postMessage' of null`. 

```js
cy.visit('https://supersuitawsstage.flqa.net')
cy.contains('h3', 'Sign In')
```

Setting `modifyObstructiveCode: false` resolves this issue. 

It's my belief we're modifying the code here - but I haven't completely verified that this is the exact point of failure - it just looked suspicious.
 
```js
function(e) {
    try {
        e = angular.module("super-templates")
    } catch (t) {
        e = angular.module("super-templates", [])
    }
    e.run(["$templateCache", function(e) {
        e.put("/silentRenew.html", "<!DOCTYPE html> <html> <head><title></title></head> <body> <script>(function(hash) {\n\tif (window.parent && window !== window.parent) {\n\t\tvar hash2 = hash || window.location.hash;\n\t\tif (hash2) {\n\t\t\twindow.parent.postMessage(hash2, location.protocol + '//' + location.host);\n\t\t}\n\t}\n})();</script> </body> </html>")
    }])
}()
```

**Comment by jennifer-shehane** [comment]
Confirmed this is not fixed in 3.1.2

**Comment by jennifer-shehane** [comment]
This now displays a different error when run, and shows up regardless of 'modifyObstructiveCode' or not:

<img width="545" alt="Screen Shot 2019-07-13 at 10 12 13 AM" src="https://user-images.githubusercontent.com/1271364/61166647-e9cff000-a556-11e9-95c2-393e7c187317.png">

Unfortunately we have to close this issue as there is not enough information to reproduce the problem. This does not mean that this issue is not happening - it just means that we do not have a path to move forward. 

Please comment in this issue with a reproducible example and we will reopen the issue. 🙏
